### ISSUE TITLE
Generated shebang with path to `python` invalid if path contains spaces

### ISSUE THREAD

**Issue by peace-maker** [Gallopsled/pwntools#2271]
Creating a venv in a path with a space causes running a process in a new terminal to fail.
```sh
$ mkdir /tmp/with\ space
$ cd /tmp/with\ space
$ python -m venv venv
$ source venv/bin/activate
$ pip install pwntools
$ python pwntools/examples/attach.py
```

Results in generating something like this
```
    #!/tmp/with space/venv/bin/python
    import os
    os.execve('/usr/bin/gdb', ['/usr/bin/gdb', '-q', '/bin/bash', '8902', '-x', '/tmp/pwnrqobb6w5.gdb'], os.environ)
```

With the shell complaining
```
bash: /tmp/tmpihzr5hsv: /tmp/with: bad interpreter: No such file or directory
```

**Comment by Arusekk** [comment]
I think the only viable workaround would be `#!/bin/env /path/with spaces/python`. But I would only emit that if there are actual spaces in the path. Would you mind crafting a PR?

**Comment by Arusekk** [comment]
Or is it just virtualenv's pip's fault? :D

**Comment by peace-maker** [comment]
Related #2285

**Comment by peace-maker** [comment]
Fixed in #2285
